src/main/kotlin/no/nav/tilleggsstonader/sak/hendelser/personhendelse/dødsfall/DødsfallHåndterer.kt

```diff
@@ -0,0 +1,137 @@
+package no.nav.tilleggsstonader.sak.hendelser.personhendelse.dødsfall
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.familie.prosessering.domene.Status
+import no.nav.familie.prosessering.domene.Task
+import no.nav.familie.prosessering.internal.TaskService
+import no.nav.tilleggsstonader.kontrakter.felles.ObjectMapperProvider.objectMapper
+import no.nav.tilleggsstonader.sak.behandling.BehandlingService
+import no.nav.tilleggsstonader.sak.fagsak.FagsakService
+import no.nav.tilleggsstonader.sak.fagsak.domain.Fagsak
+import no.nav.tilleggsstonader.sak.hendelser.Hendelse
+import no.nav.tilleggsstonader.sak.hendelser.HendelseRepository
+import no.nav.tilleggsstonader.sak.hendelser.TypeHendelse
+import no.nav.tilleggsstonader.sak.opplysninger.oppgave.OppgaveService
+import no.nav.tilleggsstonader.sak.util.erÅpen
+import no.nav.tilleggsstonader.sak.vedtak.VedtaksperiodeService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Service
+class DødsfallHåndterer(
+    private val fagsakService: FagsakService,
+    private val taskService: TaskService,
+    private val behandlingService: BehandlingService,
+    private val vedtaksperiodeService: VedtaksperiodeService,
+    private val hendelseRepository: HendelseRepository,
+    private val oppgaveService: OppgaveService,
+) {
+    companion object {
+        private const val ANNULLERT_DØDSFALL_BESKRIVELSE = "\n\nDenne oppgaven har blitt annullert"
+    }
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Transactional
+    fun håndter(dødsfallHendelse: DødsfallHendelse) {
+        if (dødsfallHendelse.erAnnullering) {
+            håndterAnnullertDødsfall(dødsfallHendelse.hendelseId)
+        } else if (hendelseRepository.existsByTypeAndId(TypeHendelse.PERSONHENDELSE, dødsfallHendelse.hendelseId)) {
+            logger.info("Dødsfallhendelse med id ${dødsfallHendelse.hendelseId} er allerede behandlet")
+        } else {
+            håndterDødsfall(dødsfallHendelse)
+        }
+    }
+
+    private fun håndterDødsfall(dødsfallHendelse: DødsfallHendelse) {
+        val fagsakerMedAktiveVedtak = finnFagsakerMedLøpendeUtbetalingFraDødsdato(dødsfallHendelse)
+
+        if (fagsakerMedAktiveVedtak.isNotEmpty()) {
+            val opprettOppgaveTask = fagsakerMedAktiveVedtak.opprettTask(dødsfallHendelse)
+
+            logger.info("Oppretter task for opprettelse av oppgave for håndtering av dødsfall som kjører ${opprettOppgaveTask.triggerTid}")
+            val lagretTask = taskService.save(opprettOppgaveTask)
+
+            hendelseRepository.insert(
+                Hendelse(
+                    type = TypeHendelse.PERSONHENDELSE,
+                    id = dødsfallHendelse.hendelseId,
+                    metadata = mapOf("taskId" to lagretTask.id.toString()),
```

Comment:
Når vi ev. skal legge til flere oppgaver hvis man har både stønader for Nay og tiltaksneheten, burde det i metadata inneholde en liste med tasks?

Reply:
Ja, hadde det slik i tidligere commit, men tok det tilbake. Men er jo selvfølgelig enklere å ha det som en liste nå 😅